### PR TITLE
Add support for CoreCLR testing via ILLINK on Linux

### PR DESCRIPTION
**Pull request by swaroop-sridhar**
Made changes to the test infrastructure to
1) Generate <test>.sh with commands to invoke ILLINK and run the output
2) Add runtest.sh --link=<path-to-illink> to invoke the ILLINK testing

### CODE REVIEW

**Comment by swaroop-sridhar** [comment]
@sbomer please review

**Comment by swaroop-sridhar** [comment]
@dotnet-bot test Windows_NT x64 Checked illink

File: tests/build_illink.cmd [hunk 1]
```diff
@@ -27,22 +26,19 @@ cd ..\linker
 ..\corebuild\Tools\dotnetcli\dotnet.exe publish -r %rid% -c netcore_Relase
 popd
```
Comment (sbomer) [inline tests/build_illink.cmd]: typo: should be netcore_Rel**e**ase

File: tests/build_illink.cmd [hunk 2]
```diff
@@ -27,22 +26,19 @@ cd ..\linker
...
 
-if not defined setEnv goto Done
-echo set ILLINK=%cd%\linker\linker\bin\netcore_Relase\netcoreapp2.0\%rid%\publish\illink.exe
-endlocal && set ILLINK=%cd%\linker\linker\bin\netcore_Relase\netcoreapp2.0\%rid%\publish\illink.exe
-
+echo Built %cd%\linker\linker\bin\netcore_Relase\netcoreapp2.0\%rid%\publish\illink.exe
```
Comment (sbomer) [inline tests/build_illink.cmd]: Rel**e**ase

File: tests/build_illink.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+pushd linker/corebuild
+./restore.sh -RuntimeIdentifier=$rid
+export DoNotEmbedDescriptors=1
+cd ../linker
+../corebuild/Tools/dotnetcli/dotnet publish -r $rid -c netcore_Release
+popd
+
+dir=$(pwd)
+output="$dir/linker/linker/bin/netcore_Relase/netcoreapp2.0/$rid/publish/illink"
```
Comment (sbomer) [inline tests/build_illink.sh]: Same typo here.

**Comment by swaroop-sridhar** [comment]
@sbomer I've made the changes you asked. 
Thanks.

**Comment by sbomer** [comment]
LGTM

**Pull request merged**